core/tee/tee_ree_fs.c

```diff
@@ -45,34 +45,52 @@
 #include <utee_defines.h>
 #include <util.h>
 
-#define BLOCK_FILE_SHIFT	12
-
-#define BLOCK_FILE_SIZE		(1 << BLOCK_FILE_SHIFT)
+/*
+ * This file implements the tee_file_operations structure for a secure
+ * filesystem based on single file in normal world.
+ *
+ * All fields in the REE file are duplicated with two versions 0 and 1. The
+ * active meta-data block is selected by the lowest bit in the
+ * meta-counter.  The active file block is selected by corresponding bit
+ * number instruct tee_fs_file_info.backup_version_table.
+ *
+ * The atomicity of each operation is ensured by updating meta-counter when
+ * everything in the secondary blocks (both meta-data and file-data blocks)
+ * are successfully written.  The main purpose of the code below is to
+ * perform block encryption and authentication of the file data, and
+ * properly handle seeking through the file. One file (in the sense of
+ * struct tee_file_operations) maps to one file in the REE filesystem, and
+ * has the following structure:
+ *
+ * [ 4 bytes meta-counter]
```

Comment:
Thanks for the description, it helps a lot. Now I understand how atomicity is enforced.
Why 4 bytes for the meta counter ? Convenience ? 1 bit would be enough I suppose, let's say 1 byte since it's the smallest addressable data size.